core/src/main/java/tc/oc/pgm/community/features/VanishManagerImpl.java

```diff
@@ -133,10 +137,38 @@ public void vanish(MatchPlayer sender, @Switch('s') boolean silent) throws Comma
   }
 
   /* Events */
+  private final Cache<UUID, String> loginSubdomains =
+      CacheBuilder.newBuilder().expireAfterAccess(1, TimeUnit.SECONDS).build();
+
+  @EventHandler(priority = EventPriority.MONITOR)
+  public void onPreJoin(PlayerLoginEvent event) {
```

Comment:
Maybe you want to do this check in `AsyncPreLoginEvent`?

Reply:
Also, before the if statement, always invalidate the `Cache` for the player.

Reply:
As discussed `AsyncPreLoginEvent` does not have a way to determine the address used to login to the server with. Sticking with `PlayerLoginEvent` for now.